api/v1/installation_types.go

```diff
@@ -141,6 +141,10 @@ type InstallationSpec struct {
 	// pods will be stuck during initialization.
 	// +optional
 	CertificateManagement *CertificateManagement `json:"certificateManagement,omitempty"`
+
+	// NonPrivileged configures Calico to be run in non privileged containers as non root users where possible.
+	// +optional
+	NonPrivileged *NonPrivilegedType `json:"nonPrivileged,omitempty"`
```

Comment:
Does this need to be a pointer? If not, then it would simplify the code elsewhere (e.g. wouldn't need to check if this was nil).

Reply:
I tried to keep this as a pointer so that a `nil` value is valid so that we can have the value omitted properly if it is empty.

Reply:
The `runAsNonPrivileged` function should treat any `nil` value as it being set to `Disabled` during the rendering. Not sure if we should explicitly set it instead of omitting the field when empty though.